src/hello-world/components/hello-world.tsx

```diff
@@ -1,32 +1,82 @@
+import { EmptyActions, EmptyState, StyledImg, Wrapper, Text } from '../styled';
 import { ChangeFirstNameModal } from './change-first-name-modal';
 import { Panel } from './control-panel';
+import { OptionsControlButton } from './options-control-button';
 import { getInlineScript, runOnloadMethod } from './script';
 
-export type DataStructure = { firstName: string; lastName: string; styles: { textAlign: string } };
+export type DataStructure = {
+  fullname: string;
+  isButtonSet: boolean;
+  styles: { textAlign: string };
+  tracking: { trackingEnabled: boolean };
+};
 
-const HelloWorld = ({ data, dispatch, className, mode }: ComponentProps<DataStructure, WithStylesProps>) => {
-  const { firstName, lastName } = data;
+const HelloWorld = ({ data, dispatch, className, mode, entityId }: ComponentProps<DataStructure, WithStylesProps>) => {
+  const { fullname, isButtonSet, tracking } = data;
   const [showModal, setShowModal] = useState(false);
+  const buttonId = `myAppButton-${entityId}`;
 
   const updateFirstName = (newFirstName: string) => {
-    dispatch((api) => api.get('firstName').set(newFirstName));
+    dispatch((api) => api.get('fullname').set(newFirstName));
     setShowModal(false);
   };
   const isViewMode = mode.type === 'view';
 
   return (
     <>
-      <div data-testid="hello-world-content" className={className}>
-        Hello,{' '}
-        <button data-testid="hello-world-first-name-btn" onClick={() => setShowModal(true)}>
-          {firstName}
-        </button>{' '}
-        {lastName}
-      </div>
+      {isButtonSet ? (
+        <>
+          <Wrapper>
+            <Text data-testid="hello-world-text">{`Your name is: ${fullname}`}</Text>
+            {/* You need to pass "className" to your button so the style changes done in "Design Settings" can be applied */}
+            <button
+              id={buttonId}
+              className={className}
+              data-testid="hello-world-change-btn"
+              onClick={() => setShowModal(true)}
+            >
+              Click here to change your name
+            </button>
+          </Wrapper>
+          {tracking.trackingEnabled && mode.type === 'publish' ? (
+            <script
+              dangerouslySetInnerHTML={{
+                __html: `
+                (function () {
+                  // Grab the form and add the conversion tracking into the submit script.
+                  const button = document.getElementById('${buttonId}');
+                  // This is a simple example of how to add conversion tracking to your app, try it on a public page by clicking the button
+                  button.onclick=function () { 
+                    ${getAfterFormSubmitScript(buttonId, true)}
+                  }
+                }())
+              `,
+              }}
+            />
+          ) : null}
+          {showModal && (
+            <ChangeFirstNameModal fullname={fullname} onUpdate={updateFirstName} onClose={() => setShowModal(false)} />
```

Comment:
Same comment as above. Use `FullNameModal` and `updateFullName` for consistency.